File: src/ReetCode/Easy/1to100/Q100/Q100_SameTree.java
/**
* Author    : Lee In Bok 
* Date      : 2023.11.14(Tue)
* Spend Time: 17m 32s
* Runtime   : 0 ms (Beats 100%)
* Memory    : 40.4 MB (Beats 06.28%)
* Algoritm  : Tree
 */

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

// class Solution {
//     public boolean ans = true;
    
//     public boolean isSameTree(TreeNode p, TreeNode q) {
//         order(p, q);

//         return ans;
//     }

//     public void order(TreeNode p, TreeNode q){
//         if(!ans) return;

//         if(p != null || q != null){
//             try {
//                 if(p.val != q.val){
//                     ans = false;
//                     return;
//                 }

//                 if(p.left != null || q.left != null) order(p.left, q.left);
//                 if(p.right != null || q.right != null) order(p.right, q.right);
//             } catch(Exception e){
//                 ans = false;
//             }
//         }
//     }
// }

File: src/ReetCode/Easy/1to100/Q100/Q100_Same Tree.py
"""
# Author    : Lee In Bok 
# Date      : 2023.11.14(Tue)
# Spend Time: 17m 32s
# Runtime   : 34 ms (Beats 81.70%)
# Memory    : 16.4 MB (Beats 20.68%)
# Algoritm  : Tree
"""

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# class Solution:
#     def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
#         self.ans = True

#         def order(p, q):
#             if not self.ans:
#                 return 

#             if p or q:
#                 try :
#                     if p.val != q.val:
#                         self.ans = False
#                         return

#                     if p.left or q.left:
#                         order(p.left, q.left)
#                     if p.right or q.right:
#                         order(p.right, q.right)
#                 except:
#                     self.ans = False

#         order(p, q)
#         return self.ans